pyanaconda/ui/lib/subscription.py

```diff
@@ -296,12 +310,19 @@ def register_and_subscribe(payload, progress_callback=None, error_callback=None,
     # - at the moment this is true only for the CDROM source
     source_proxy = payload.get_source_proxy()
     if payload.type == PAYLOAD_TYPE_DNF:
+        switched_source = False
         if source_proxy.Type in SOURCE_TYPES_OVERRIDEN_BY_CDN:
             log.debug("subscription thread: overriding current installation source by CDN")
             switch_source(payload, SOURCE_TYPE_CDN)
+            switched_source = True
         # If requested, also restart the payload if CDN is the installation source
-        # The CDN either already was the installation source or we just switched to it.
-        if restart_payload and source_proxy.Type == SOURCE_TYPE_CDN:
+        # or if the source was switched.
+        #
+        # If the source was just switched, then source_proxy.Type will not be updated
+        # until payload refresh has been done. For this reason we also need to check
+        # the value of source_switched variable and restart payload if it is true,
+        # even if CDN does not appear to be the current installation source.
+        if restart_payload and (switched_source or source_proxy.Type == SOURCE_TYPE_CDN):
```

Comment:
The statement `source_proxy.Type` doesn't return a correct type, because `source_proxy` is a proxy of the previous source. You should call `source_proxy = payload.get_source_proxy()` to work with the new source.

Reply:
Oh, so that's why it happens! Still, I wonder if we want to change the code ? Or maybe just fix the documentation but keep the logic, as that has already been tested by RTT & appears to be working fine ?

Reply:
I would prefer correct fix here.

Reply:
Hmm, looking into it in more detail, it really looks better, let's do that. :)

Reply:
Switched to the better fix. :)